lib/Parser/Parser.cpp

```diff
@@ -46,7 +47,12 @@ Optional<PredicateDecl> Parser::parsePredicateDecl() {
             if(parseParameter().unwrapInto(param)) {
                 parameters.push_back(param);
             } else {
-                emitSyntaxError("Expected an additional argument after \",\" in parameter list.");
+                if (parameters.size() == 0) {
+                    emitSyntaxError("Parentheses must not appear after predicate name for predicates with zero arguments.");
```

Comment:
Text of the new error message is up for debate

Reply:
I like the second text better, since it answers the question of "where?" better. My rule of thumb is that new users of the language are the best people to evaluate diagnostic messages, so I actually trust you more than me here 😄